hyperspy/_signals/eels.py

```diff
@@ -1433,6 +1487,197 @@ def create_model(self, ll=None, auto_background=True, auto_add_edges=True,
+        # get corresponding information depending on whether it is an element
+        # a particular edge or a family of edge
+        axis_min = self.axes_manager[-1].low_value
+        axis_max = self.axes_manager[-1].high_value
+        
+        names_and_energies = {}
+        shells = ["K", "L", "M", "N", "O"]
+        
+        errmsg = ("Edge family '{}' is not supported. Supported edge family "
+                  "is {}.")
+        for member in element_edge_family:
+            try:
+                element, ss = member.split("_")
+                
+                if len(ss) == 1:
+                    memtype = 'family'
+                    if ss not in shells:
+                        raise AttributeError(errmsg.format(ss, shells))            
+                if len(ss) == 2:
+                    memtype = 'edge'
+                    if ss[0] not in shells:
+                        raise AttributeError(errmsg.format(ss[0], shells))                           
+            except ValueError:
+                element = member
+                ss = ''
+                memtype = 'element'
+
+            try:
+                Binding_energies = elements_db[element]["Atomic_properties"]["Binding_energies"]
+            except KeyError as err:
+                raise ValueError("'{}' is not a valid element".format(element)) from err
+                
+            for edge in Binding_energies.keys(): 
+                relevance = Binding_energies[edge]["relevance"]
+                energy = Binding_energies[edge]["onset_energy (eV)"]
+                
+                isInRel = relevance in only_edges
+                isInRng = axis_min < energy < axis_max 
+                isSameFamily = ss in edge
+                
+                if memtype == 'element': 
+                    flag = isInRel & isInRng
+                    edge_key = element + "_" + edge
+                elif memtype == 'edge':
+                    flag = isInRng & (edge == ss)
+                    edge_key = member
+                elif memtype == 'family':
+                    flag = isInRel & isInRng & isSameFamily
+                    edge_key = element + "_" + edge
+                
+                if flag:
+                    names_and_energies[edge_key] = energy
+                    
+        return names_and_energies
+
+    def _edge_marker_closed(self, obj):
+        marker = obj
+        for EELS_edge, line_markers in reversed(list(
+                self._edge_markers.items())):
+            if marker in line_markers:
+                line_markers.remove(marker)
+            if not line_markers:
+                self._edge_markers.pop(EELS_edge)
+
+    def remove_EELS_edges_markers(self, EELS_edges):
+        for EELS_edge in EELS_edges:
+            if EELS_edge in self._edge_markers:
+                line_markers = self._edge_markers[EELS_edge]
+                while line_markers:
+                    m = line_markers.pop()
+                    m.close()
```

Comment:
It may be worth considering using the argument `render_figure` to speed up removing marker when using matplotlib backend supporting blitting but let sort out first the `_clear_all_markers` bit.

Reply:
`render_figure` is set `False` now whenever markers are added or closed.